docs/lesson-plans/week-7.md

```diff
@@ -0,0 +1,692 @@
+working with is called a relational database, which means it acts like a collection of Excel sheets. It also has a nice
+little syntax for performing operations called SQL (pronounced "see-quel" or "ess-que-ell"). There are also 
+other types of DBs, called NoSQL databases, which act a little differently, but we'll just stick with relational in this
+program.
+
+### Excel Sheets to Database Tables
+
+When you think of Excel, you think of a bunch of rows/entries and columns/fields which make up a sheet. In terms of DBs, 
+each table is like a separate sheet containing pretty uniform data. Each column (the up/down ones) acts like a separate 
+field. You specify a datatype and name for your columns, and then as data gets entered, rows (the long left/right) ones 
+get created, read, updated, and deleted (remember CRUD). One thing to note is that in relational databases, unlike 
+Excel, you don't have an unlimited amount of columns that can be automatically added. When you create your table, it has 
+a set structure, meaning you can only add/remove/update columns by changing the structure of the table. You can, 
+however, add an unlimited number of rows in theory, though you'll run out of space and your queries will start to slow 
+down after a while. 
+
+Here's a reminder of what an Excel sheet can look like:
+
+| These | are | your | columns | :) | 
+| :---: | :---: | :---: | :---: | :---: |
+| T | h | e | s | e |
+| a | r | e | :) | :) |
+| y | o | u | r | :) |
+| r | o | w | s | :) |
+
+### Data Types in SQL
+
+In SQL, when making new columns on a table, altering columns, and dealing with table data, you need to be aware of the
+data types of the variables you're working with. It's somewhat similar to working with types in Java, however, you don't
+need to declare types all over the place. It matters most when you're trying to perform operations on types and getting 
+the data using the JDBC/ODBC. For example, if you try to perform a query where you're adding two strings, you'll receive 
+an error since strings can't be added. You can concatenate two strings, but addition is for number types. 
+
+There are many types available, especially in PostgreSQL, but below are a few of the most commonly used ones.
+
+#### SERIAL
+
+The `SERIAL` type is an `INTEGER` under the hood, usually used as an ID. It's a 
+PostgreSQL specific data type, which is usually just an `INTEGER` with the `AUTO INCREMENT` modifier in most other 
+databases. This is really useful for integer-typed
+`PRIMARY KEYS` in other databases, since it automatically creates a new unique ID for you.
+
+#### CHAR
+
+The `CHAR(x)` type is actually a string of max length `x`. If a value you insert isn't the full length `x`, then the 
+value will be right-padded with spaces. For example, if the column type is `CHAR(5)`, then "hello" would be saved as 
+"hello", but "hey" would be saved as "hey__" (pretend the underscores are spaces).
```

Comment:
I think its fine to just put "hey  " instead of "hey__", I'm thinking it can possible be misinterpreted by thinking "hey__" will be saved. I know you say "pretend the underscores are spaces" but I think its fine to just put "hey  " and say something like "(Notice how there is are two spaces at the end of the string)".

Reply:
This itself is totally fine too but it was just a thought!

Reply:
Makes sense! I wanted to emphasize that they were different, but I'm sure the __ can be confusing/overkill